### DISCUSSION TITLE
Convert array of unknown parameters from json into a final string

### DISCUSSION THREAD

**Discussion by AndreAhmed** [nlohmann/json#4006] [Q&A]
I have the following json samples:
I would like to get all those parameters inside the json, the key, value as strings
and convert them to a string

@nlohmann 

```
{
    "parameters": {
        "commandBuffer": "0x0000000000012ad2",
        "indexCount": 156,
        "instanceCount": 1,
        "firstIndex": 0,
        "vertexOffset": 0,
        "firstInstance": 0
    }
}
```
and

```
{
    "parameters": {
        "commandBuffer": "0x0000000000012ad2",
        "pipelineBindPoint": "VK_PIPELINE_BIND_POINT_GRAPHICS",
        "layout": "0x000000000000f42e",
        "firstSet": 0,
        "descriptorSetCount": 1,
        "pDescriptorSets": [
            "0x000000000001411e"
        ],
        "dynamicOffsetCount": 1,
        "pDynamicOffsets": [
            4589632
        ]
    }
}
```
As you see in the Json, there always be a field called parameters, but with multiple fields that are unknown at runtime.

I would like to know a way, to convert those objects into a useful string, giving unknown parameters list.

**Comment by nlohmann** [comment]
Since you don't know the keys ahead of time, iterate the "parameters" object with .items() and use dump() on each value to get a textual representation regardless of its type: std::string out; for (auto& [key, value] : j["parameters"].items()) { out += key + "=" + (value.is_string() ? value.get<std::string>() : value.dump()) + ";"; }. value.dump() works uniformly for numbers, strings, arrays, booleans, etc. (numbers/bools/arrays get their JSON text form; strings would get quoted unless you special-case is_string() as above to avoid embedded quotes).

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
